evdns.c

```diff
@@ -5564,8 +5564,9 @@ evdns_getaddrinfo_fromhosts(struct evdns_base *base,
 	EVDNS_UNLOCK(base);
 out:
 	if (n_found) {
-		/* Note that we return an empty answer if we found entries for
-		 * this hostname but none were of the right address type. */
+		if (!ai) {
+			return EVUTIL_EAI_ADDRFAMILY;
```

Comment:
But this error does not propagated to the caller, it simply means that "OK, let's ignore entries in the /etc/hosts if they do not match"